src/flamenco/runtime/program/fd_vote_program.c

```diff
@@ -488,63 +492,69 @@ is_uninitialized( fd_vote_state_versioned_t * self ) {
 
 // https://github.com/firedancer-io/solana/blob/da470eef4652b3b22598a1f379cacfe82bd5928d/sdk/program/src/vote/state/vote_state_versions.rs#L15
 static void
-convert_to_current( fd_vote_state_versioned_t * self, fd_exec_instr_ctx_t const * ctx ) {
+convert_to_current( fd_vote_state_versioned_t * self,
+                    fd_valloc_t                 valloc ) {
   switch( self->discriminant ) {
   // https://github.com/firedancer-io/solana/blob/da470eef4652b3b22598a1f379cacfe82bd5928d/sdk/program/src/vote/state/vote_state_versions.rs#L17-L50
   case fd_vote_state_versioned_enum_v0_23_5: {
-    fd_vote_state_0_23_5_t      state = self->inner.v0_23_5;
+    fd_vote_state_0_23_5_t * state = &self->inner.v0_23_5;
     fd_vote_authorized_voters_t authorized_voters;
     authorized_voters_new(
-        state.authorized_voter_epoch, &state.authorized_voter, ctx, &authorized_voters );
+        state->authorized_voter_epoch, &state->authorized_voter, valloc, &authorized_voters );
 
     /* Temporary to hold current */
-    fd_vote_state_t current;
-    current.node_pubkey           = state.node_pubkey;
-    current.authorized_withdrawer = state.authorized_withdrawer;
-    current.commission            = state.commission;
-    current.votes                 = landed_votes_from_lockouts( state.votes, ctx );
-    current.root_slot             = state.root_slot;
-    current.authorized_voters     = authorized_voters;
-    current.prior_voters          = ( fd_vote_prior_voters_t ){
-                 .idx      = 31UL,
-                 .is_empty = 1,
+    fd_vote_state_t current = {
+      .node_pubkey           = state->node_pubkey,            /* copy */
+      .authorized_withdrawer = state->authorized_withdrawer,  /* copy */
+      .commission            = state->commission,             /* copy */
+      .votes                 = landed_votes_from_lockouts( state->votes, valloc ),
+      .root_slot             = state->root_slot,  /* deep copy */
+      .authorized_voters     = authorized_voters,
+      .prior_voters = (fd_vote_prior_voters_t) {
+        .idx      = 31UL,
+        .is_empty = 1,
+      },
+      .epoch_credits  = state->epoch_credits,   /* move */
+      .last_timestamp = state->last_timestamp,  /* deep copy */
     };
-    memset( current.prior_voters.buf, 0, sizeof( current.prior_voters.buf ) );
-    current.epoch_credits  = state.epoch_credits;
-    current.last_timestamp = state.last_timestamp;
+
+    /* Move objects */
```

Comment:
Just a thought: maybe we should have an `fd_move` which sets the old location to `NULL`?